tests/testthat/test-validate.R

```diff
@@ -0,0 +1,322 @@
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+  expect_error(
+    mighty_component$new(template_bad_param, "A"),
+    "Invalid.*@param.*Must have both name and description"
+  )
+})
+
+test_that("multi-line @param description works", {
+  # Missing description
+  template_bad_param <- c(
+    "#' @title Test Component",
+    "#' @description Test description", 
+    "#' @param variable long multi line description that
+    #' goes",
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+
+  expect_no_error(
+    mighty_component$new(template_bad_param, "A"))
+})
+
+test_that("@depends tags needs both domain and column", {
+  # Missing column
+  template_bad_depends <- c(
+    "#' @title Test Component",
+    "#' @description Test description",
+    "#' @depends .self",
+    "#' @type derivation",
+    "#' @code", 
+    ".self"
+  )
+  
+  expect_error(
+    validate_template(template_bad_depends),
+    "Invalid.*@depends.*Must have both domain and column"
+  )
+  
+  # Valid depends format
+  template_valid_depends <- c(
+    "#' @title Test Component",
+    "#' @description Test description",
+    "#' @depends .self USUBJID",
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+  
+  expect_no_error(validate_template(template_valid_depends))
+})
+
+test_that("validate_template validates @outputs tags", {
+  template_valid_output <- c(
+    "#' @title Test Component",
+    "#' @description Test multile",
+    "#' description",
+    "#' @outputs NEWVAR",
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+  
+  expect_no_error(validate_template(template_valid_output))
+})
+
+test_that("duplicate tags", {
+  template_valid_output <- c(
+    "#' @title Test Component",
+    "#' @title Test Component",
+    "#' @description Test desc",
+    "#' @outputs NEWVAR",
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+  
+  expect_error(validate_template(template_valid_output), "Multiple or no matches found for tag: '@title'")
+})
+
+test_that("validate_template handles complex valid template", {
+  complex_template <- c(
+    "#' @title Complex Component",
+    "#' @description",
+    "#' This is a complex component that does multiple things",
+    "#' across multiple lines of description.",
+    "#'", 
+    "#' @param variable Name of output variable",
+    "#' @param input_var Name of input variable",
+    "#' @type derivation",
+    "#' @depends .self {{input_var}}",
+    "#' @depends .self USUBJID",
+    "#' @outputs {{variable}}",
+    "#' @outputs var_B",
+    "#' @code",
+    ".self <- .self |>",
+    "  dplyr::mutate(",
+    "    {{variable}} = case_when(",
+    "      {{input_var}} > 0 ~ 'positive',",
+    "      {{input_var}} < 0 ~ 'negative',", 
+    "      TRUE ~ 'zero'",
+    "    )",
+    "  )"
+  )
+  
+  expect_no_error(validate_template(complex_template))
+})
+
+test_that("validate_template rejects empty outputs tag", {
+  empty_outputs <- c(
+    "#' @title empyt code",
```

Comment:
Typo in test data: 'empyt' should be 'empty'
```suggestion
    "#' @title empty code",
```